js/components/controlBar.js

```diff
@@ -219,6 +221,60 @@ var ControlBar = React.createClass({
     this.props.controller.state.focusedControl = null;
   },
 
+  /**
+   * Will handle the keydown event when the controlBar is active and it will restrict
+   * tab navigation to elements that are within it when the player is in fullscreen mode.
+   * Note that this only handles the edge cases that are needed in order to loop the tab
+   * focus. Tabbing in between the elements is handled by the browser.
+   * @private
+   * @param {Object} evt Keydown event object.
+   */
+  handleControlBarKeyDown: function(evt) {
+    if (
+      evt.key !== CONSTANTS.KEY_VALUES.TAB ||
+      !this.props.controller.state.fullscreen ||
+      !this.domNode ||
+      !evt.target
+    ) {
+      return;
+    }
+    // Focusable elements on the control bar (this.domNode) are expected to have the
+    // data-focus-id attribute
+    var focusableElements = this.domNode.querySelectorAll('[data-focus-id]');
+
+    if (focusableElements.length) {
+      var firstFocusableElement = focusableElements[0];
+      var lastFocusableElement = focusableElements[focusableElements.length - 1];
```

Comment:
Is it possible for these first and last elements to be opposite in terms of tab order that the browser handles? Like the "last" focusable element precedes the first when just normally tabbing. 

Also, do we need any safeguards that the user isn't stuck tabbing between the first and last element only?

Reply:
I haven't found any official confirmation, but `querySelectorAll` uses depth-first pre-order traversal and it seems that the regular tab navigation flow (with tabindex=0) does this as well. At least according to this: https://goo.gl/8pF94w
If this is true then the element order will always be the same as the tab order. I did try this in all browsers though, and it works fine.

Reply:
Sounds good. Thanks for the thorough check!